### DISCUSSION TITLE
type hint issue for rename in pandas

### DISCUSSION THREAD

**Discussion by stevenlis** [microsoft/pylance-release#3630] [Q&A]
I don't quite understand the warning info and not sure where to report this

```python
import pandas as pd
data = {'col1': [1, 3, 4], 'col2': [2, 3, 5], 'col2': [2, 4, 4]}
df = pd.DataFrame(data)

df.rename(columns={'col1': 'col'})  # no warning

df.groupby(['col1'], as_index=False).size().rename(columns={'size': 'n'})  # warning
```
![CS-G3RWj1iN](https://user-images.githubusercontent.com/33796896/201777168-4a84d4f0-59dd-45e2-9f8c-c584933b58d3.png)

![CS-wVdY9LAU](https://user-images.githubusercontent.com/33796896/201777075-24d53c7e-6e1c-4e39-9caa-94090213995b.png)

**Comment by debonte** [comment]
The first `rename` call is to `DataFrame.rename`. The second is to `Series.rename` which takes different arguments. In particular, it doesn't have a `column` argument.

**Comment by stevenlis** [comment]
@debonte I'm not sure that's ture since I set `as_index=False`.
![CS-56cF2LkJ](https://user-images.githubusercontent.com/33796896/201801627-238adf2b-baee-4856-bb50-6fc2d3bf37e2.png)

**Reply by debonte** [reply]
Ok, I see that [the docs](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.core.groupby.GroupBy.size.html) say that. However, the stubs from [pandas-stubs](https://github.com/pandas-dev/pandas-stubs/blob/main/pandas-stubs/core/groupby/groupby.pyi#L120) (which we ship with Pylance) claim that `size` always returns `Series[int]`.

Based on [this comment](https://github.com/pandas-dev/pandas-stubs/issues/203#issuecomment-1218434654), this issue appears to be tracked by https://github.com/pandas-dev/pandas-stubs/issues/203.

**Reply by stevenlis** [reply]
@debonte  thanks man. it's good to know that it's indeed a bug.
